sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -5,15 +5,17 @@
 #define AZ_CURL_ADAPTER_H
 
 #include <az_callback.h>
-#include <az_http_request.h>
-#include <az_span_seq.h>
+#include <az_http_request_builder.h>
+#include <az_pair.h>
 #include <az_span_builder.h>
+#include <az_span_seq.h>
+#include <az_str.h>
+
+#include <_az_cfg_prefix.h>
```

Comment:
Can you take the headers order from master branch? It has _az_cfg included in the right place, and don't have unused headers, such as az_str.